src/mbgl/programs/segment.hpp

```diff
@@ -37,7 +36,7 @@ class Segment {
     //     data-driven paint properties
     //   * when two fill layers have the same layout properties, but one
     //     uses fill-color and the other uses fill-pattern
-    mutable std::map<std::string, gfx::DrawScope> drawScopes;
+    mutable std::shared_ptr<std::map<std::string, gfx::DrawScope>> drawScopes;
```

Comment:
why is this a `shared_ptr`? Segments should be unique

Reply:
Also, if we hide this behind a pointer, we can remove the `mutable` since the indirection makes them mutable automatically.

Reply:
I need to sort symbol segments across the tiles. Initially I had https://github.com/mapbox/mapbox-gl-native/pull/14409/commits/fa37b21d8fd13e5f5afdb4062d0b9f2aa3f681e1#diff-e28696fa5b68ae9576f684ddfdbaf98cR39 vector of reference wrappers, however, that added new template parameter for draw => new template instantiation.

`shared_ptr` makes `Segment` copyable, therefore, I can create vector of segments.